contracts/core/Core.sol

```diff
@@ -219,4 +234,87 @@ contract Core is
             _quantity
         );
     }
+
+    /**
+     * Function to convert Set Tokens into underlying components
+     *
+     * @param _tokenAddress    The address of the Set token
+     * @param _quantity        The number of tokens to redeem
+     */
+    function redeem(
+        address _tokenAddress,
+        uint _quantity
+    )
+        public
+        isValidTokenAddress(_tokenAddress)
+        isNonZero(_quantity)
+    {
+        // Check if multiple of naturalUnit
+        uint naturalUnit = SetToken(_tokenAddress).naturalUnit();
+        require(isMultipleOfNaturalUnit(_quantity, naturalUnit));
+
+        // Burn the Set token
+        SetToken(_tokenAddress).burn(_quantity);
+
+        // Decrement the Set Token amount (this will validate if balance has enough)
+        Vault(vaultAddress).decrementTokenOwner(
```

Comment:
Set Token balances aren't kept in Vault but in the Set Token contract, burning the token changes the Set Token balance. Don't need this line of code.

Reply:
ah okay so need to just make decrements to the component values in Vault

Reply:
We should be incrementing the components in the Vault and decrementing (through burn) the Set Token in the SetToken contract if that makes sense. It's basically debiting one account (SetToken) and crediting another (Vault). Purely internal accounting, then if someone wants to withdraw their tokens they could call withdraw in Core and thats how they'd get the constituent tokens back.

Reply:
For SetToken.burn...it currently burns from the `msg.sender` balance. do you know if that is going to be `Core.sol` in this case or the `msg.sender` who first called the `redeem` method?

Reply:
It would be msg.sender because the function is being called in Core, so the parameter that is being passed is the msg.sender address as it related to Core. Inside the SetToken contract msg.sender should not be referenced

Reply:
so the burn function should have another parameter that passes on the address to burn the token from (in this case msg.sender). Make sense?